src/flag_gems/ops/index_add.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+import torch
+import triton
+import triton.language as tl
+
+from ..utils import dim_compress, libentry
+
+
+def cfggen():
+    block_m = [1, 2, 4, 8]
+    configs = [
+        triton.Config({"BLOCK_M": m, "BLOCK_N": 1024}, num_warps=4) for m in block_m
+    ]
+    return configs
+
+
+@libentry()
+@triton.autotune(configs=cfggen(), key=["M", "N"])
+@triton.jit
+def index_add_kernel(
+    inp,
+    out,
+    index,
+    src,
+    M,
+    N,
+    alpha,
+    inp_len,
+    BLOCK_M: tl.constexpr,
+    BLOCK_N: tl.constexpr,
+):
+    pid = tl.program_id(0)
+    rows_offsets = pid * BLOCK_M + tl.arange(0, BLOCK_M)[:, None]
+    rows_mask = rows_offsets < M
+
+    for off in range(0, N, BLOCK_N):

```

Comment:
Maybe a 2d grid can be used, since if `M` is small and `N` is large, this task partition scheme may not saturate a gpu's SMs.

Reply:
> Maybe a 2d grid can be used, since there is no dependencies between iterations in the loop. If `M` is small and `N` is large, this task partition scheme may not saturate a gpu's SMs.

! I get what you mean now; that does make sense. I'll solve this part.